src/dc_receive_imf.rs

```diff
@@ -136,14 +119,13 @@ pub fn dc_receive_imf(
         if from_ids.len() > 1 {
             warn!(
                 context,
-                "mail has more than one address in From: {:?}", field_from
+                "mail has more than one From address, only using first: {:?}", field_from
             );
         }
         if from_ids.contains(&DC_CONTACT_ID_SELF) {
             incoming = false;
-            if to_ids.len() == 1 && to_ids.contains(&DC_CONTACT_ID_SELF) {
-                from_id = DC_CONTACT_ID_SELF;
-            }
+            from_id = DC_CONTACT_ID_SELF;
+            incoming_origin = Origin::OutgoingBcc;
```

Comment:
not sure why we had this weird condition before.  If we are in the From, then from_id is DC_CONTACT_ID_SELF independent from what is in to_ids.